sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -537,7 +537,17 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
       rawTable.storage
     }
 
-    val newTable = rawTable.copy(
+    // Fills in Hive provider if necessary. Otherwise, `Hive.alterTable` will overwrite
+    // important table metadata like bucketing when altering table.
+    val tableMeta = rawTable.properties.get(DATASOURCE_PROVIDER) match {
+      // This is a Hive serde table, set its provider.
+      case None if rawTable.tableType != VIEW =>
```

Comment:
Does it mean this table is created by Hive?

Reply:
This just means the table is Hive serde table. I think we can also use Spark command to create Hive table.